scripts/generators/beats.py

```diff
@@ -10,10 +10,17 @@ def generate(ecs_nested, ecs_version, out_dir):
     # base first
     beats_fields = fieldset_field_array(ecs_nested['base']['fields'], df_whitelist, ecs_nested['base']['prefix'])
 
+    # tracing fields also need top-level handling
+    tracing_fields = fieldset_field_array(
+        ecs_nested['tracing']['fields'], df_whitelist, ecs_nested['tracing']['prefix'])
+    beats_fields.extend(tracing_fields)
+
     allowed_fieldset_keys = ['name', 'title', 'group', 'description', 'footnote', 'type']
+    fieldsets_to_skip = ['base', 'tracing']
```

Comment:
The special handling for `base` here had two purposes. One of which was putting the fields at the root rather than nested, just like we're fixing for the tracing fields. But the other purpose was to literally [start the file](https://github.com/elastic/ecs/blob/master/generated/beats/fields.ecs.yml#L9-L45) with the base fields. Just like we display "Base" first in the list of field sets, in the documentation.

In order to correctly handle future field sets whose fields are also at the root, I think we should use the field set's `root` attribute. It's set to true only for field sets that shouldn't be nested.

So concretely, I would leave the special handling for `base` as it was, to keep the fields first in the file.

But in the loop we could test for `if fieldset['root']`, then either `.extend` or copy.

Note: I think we should try to avoid `root:true` field sets in general, because they cause confusion. But that doesn't mean they're not going to happen again :-)

Reply:
This approach seems sensible since we already have the attribute present and set to true for these fields. 😄 

I've adjusted here.